internal/envoy/version_test.go

```diff
@@ -48,7 +47,9 @@ func TestGetHomeVersion_Empty(t *testing.T) {
 		{"missing home version file", func() {
 			require.NoError(t, os.Remove(homeVersionFile))
 		}},
-		{"missing home dir", removeHomeDir},
+		{"missing home dir", func() {
+			require.NoError(t, os.Remove(homeDir))
```

Comment:
nit do we need this? I believe homeDir is cleaned up internally by testing.T.

Reply:
good question. in this case it isn't about cleanup, rather that this is testing behavior when it isn't there. Since most tests need the directory, it is eagerly provisioned and only cleaned up for this case